src/Workspaces/Core/Portable/Workspace/IWorkspaceConfigurationService.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+using Microsoft.CodeAnalysis.Storage;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    internal interface IWorkspaceConfigurationService : IWorkspaceService
+    {
+        WorkspaceConfigurationOptions Options { get; }
+    }
+
+    /// <summary>
+    /// Options that affect behavior of workspace core APIs (<see cref="Solution"/>, <see cref="Project"/>, <see cref="Document"/>, <see cref="SyntaxTree"/>, etc.)
+    /// to which it would be impractical to flow these options explicitly. The options are instead provided by <see cref="IWorkspaceConfigurationService"/>.
+    /// The remote instance of this service is initialized based on the in-proc values (which themselves are loaded from global options) when we establish connection
+    /// from devenv to ServiceHub process. If another process connects to our ServiceHub process before that the reote instance provides a predefined set of options
+    /// <see cref="RemoteDefault"/> that can later be updated when devenv connects to the ServiceHub process.
+    /// </summary>
+    [DataContract]
+    internal readonly record struct WorkspaceConfigurationOptions(
+        [property: DataMember(Order = 0)] StorageDatabase CacheStorage = StorageDatabase.SQLite,
+        [property: DataMember(Order = 1)] bool DisableRecoverableTrees = false,
+        [property: DataMember(Order = 2)] bool DisableProjectCacheService = false,
+        [property: DataMember(Order = 3)] bool EnableOpeningSourceGeneratedFiles = false)
+    {
+        public WorkspaceConfigurationOptions()
+            : this(CacheStorage: StorageDatabase.SQLite)
+        {
+        }
+
+        public static readonly WorkspaceConfigurationOptions Default = new();
+
+        /// <summary>
+        /// These values are such that the correctness of remote services is not affected if these options are changed from defaults
+        /// to non-defauls while the services have already been executing.
+        /// </summary>
+        public static readonly WorkspaceConfigurationOptions RemoteDefault = new(
+            CacheStorage: StorageDatabase.None,
```

Comment:
oh, what's going on here?  why is Remote set to StorageDB.None?

Reply:
Explained here: https://github.com/dotnet/roslyn/pull/59790/files/06917a20d8590b38138091382e49715a1e1456f1#diff-428c301ae3b036a3bfab4419b4be5a0e0815ec9b52cd4937439584caf64ae663R19-R20

Essentially, we want to avoid using the cache until the process (oop) is initialized with the right configuration sent from devenv.